samples/AuthorizationServerExample/Program.cs

```diff
@@ -0,0 +1,135 @@
+                        CallToolHandler = (request, cancellationToken) =>
+                        {
+                            if (request.Params?.Name == "echo")
+                            {
+                                if (request.Params.Arguments?.TryGetValue("message", out var message) is not true)
+                                {
+                                    throw new McpException("Missing required argument 'message'");
+                                }
+
+                                return new ValueTask<CallToolResponse>(new CallToolResponse()
+                                {
+                                    Content = [new Content() { Text = $"Echo: {message}", Type = "text" }]
+                                });
+                            }
+                            
+                            // Protected tool that requires authorization
+                            if (request.Params?.Name == "protected-data")
+                            {
+                                // This tool will only be accessible to authenticated clients
+                                return new ValueTask<CallToolResponse>(new CallToolResponse()
+                                {
+                                    Content = [new Content() { Text = "This is protected data that only authorized clients can access" }]
+                                });
+                            }
+
+                            throw new McpException($"Unknown tool: '{request.Params?.Name}'");
+                        },
+                        
+                        ListToolsHandler = async (_, _) => new()
+                        {
+                            Tools = 
+                            [
+                                new() 
+                                { 
+                                    Name = "echo", 
+                                    Description = "Echoes back the message you send" 
+                                },
+                                new()
+                                {
+                                    Name = "protected-data",
+                                    Description = "Returns protected data that requires authorization"
+                                }
+                            ]
+                        }
+                    }
+                };
+            })
+            .WithAuthorization(authProvider)  // Enable authorization with our provider
+            .WithHttpTransport();             // Configure HTTP transport
+
+        var app = builder.Build();
+        
+        // 5. Enable authorization middleware (this must be before MapMcp)
+        // This middleware does several things:
+        // - Serves the PRM document at /.well-known/oauth-protected-resource
+        // - Checks Authorization header on requests
+        // - Returns 401 + WWW-Authenticate when authorization is missing or invalid
+        app.UseMcpAuthorization();
```

Comment:
This PR introduces custom implementations for authorization, which reimplements much of the functionality already provided by ASP.NET Core's built-in authorization middleware. Leveraging the standard MapX().RequireAuthorization() approach would not only simplify the implementation for library consumers but also align with best practices.

Could you re-evaluate this implementation and consider using MapMcp().RequireAuthorization() instead of custom solutions for handling authorization? This would ensure better maintainability, security, and alignment with typical ASP.NET Core practices.

Reply:
Doing so would allow for usage with all the builtin integrations and libraries that already support asp.net core authentication.

Reply:
Put a bit differently, if you want everyone inside and outside of msft to use this library, the code needs to meet them where they already are rather than forcing something new that has similar type names and concepts.